packages/headless/src/api/analytics/instant-result-analytics.ts

```diff
@@ -0,0 +1,23 @@
+import {SearchAnalyticsProvider} from './search-analytics';
+
+export class InstantResultsAnalyticsProvider extends SearchAnalyticsProvider {
+  private get instantResultsSearchUid() {
+    const state = this.getState().instantResults;
```

Comment:
Is there a reasonable scenario where that property would be undefined ?

When we instantiate that class, in `instant-result-analytics-action`, is the `getState()` return typed properly such that we know it's always defined ?

If so, we could have a constructor that enforce the proper type.

Reply:
Done ✔️